lucene/core/src/java/org/apache/lucene/util/hnsw/HnswGraphSearcher.java

```diff
@@ -70,6 +72,43 @@ public static void search(
     search(scorer, knnCollector, graph, graphSearcher, acceptOrds);
   }
 
+  /**
+   * Searches the HNSW graph for for the nerest neighbors of a query vector, starting from the
+   * provided entry points.
+   *
+   * @param scorer the scorer to compare the query with the nodes
+   * @param knnCollector a collector of top knn results to be returned
+   * @param graph the graph values. May represent the entire graph, or a level in a hierarchical
+   *     graph.
+   * @param acceptOrds {@link Bits} that represents the allowed document ordinals to match, or
+   *     {@code null} if they are all allowed to match.
+   * @param entryPointOrds the entry points for search.
+   */
+  public static void search(
+      RandomVectorScorer scorer,
+      KnnCollector knnCollector,
+      HnswGraph graph,
+      Bits acceptOrds,
+      DocIdSetIterator entryPointOrds)
+      throws IOException {
+    ArrayList<Integer> entryPointOrdInts = new ArrayList<Integer>();
+    if (entryPointOrds != null) {
```

Comment:
maybe, to avoid `entryPointOrdInts` object allocation if `entryPointOrds` is null.
```suggestion
    ArrayList<Integer> entryPointOrdInts = null;
    if (entryPointOrds != null) {
      entryPointOrdInts = new ArrayList<Integer>();
```